src/pyqasm/visitor.py

```diff
@@ -86,6 +86,7 @@ def __init__(  # pylint: disable=too-many-arguments
         external_gates: list[str] | None = None,
         unroll_barriers: bool = True,
         max_loop_iters: int = int(1e9),
+        device_qubits: int = 0,
```

Comment:
Derive from `module`, use something like a `_consolidate_qubits` flag to check and perform the consolidation